kernel/kernel-api/src/test/scala/io/delta/kernel/internal/DeltaHistoryManagerSuite.scala

```diff
@@ -23,25 +23,51 @@ import scala.collection.JavaConverters._
 import scala.reflect.ClassTag
 
 import io.delta.kernel.TransactionSuite.testSchema
+import io.delta.kernel.data.{ColumnarBatch, ColumnVector}
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.exceptions.KernelException
 import io.delta.kernel.exceptions.TableNotFoundException
 import io.delta.kernel.internal.actions.{Format, Metadata, Protocol}
 import io.delta.kernel.internal.commit.DefaultFileSystemManagedTableOnlyCommitter
+import io.delta.kernel.internal.files.ParsedLogData
 import io.delta.kernel.internal.fs.Path
 import io.delta.kernel.internal.lang.Lazy
 import io.delta.kernel.internal.metrics.SnapshotQueryContext
 import io.delta.kernel.internal.snapshot.LogSegment
 import io.delta.kernel.internal.util.{FileNames, VectorUtils}
 import io.delta.kernel.internal.util.InCommitTimestampUtils
 import io.delta.kernel.internal.util.VectorUtils.{buildArrayValue, stringStringMapValue}
-import io.delta.kernel.test.MockFileSystemClientUtils
+import io.delta.kernel.test.{MockFileSystemClientUtils, MockListFromFileSystemClient, MockReadICTFileJsonHandler}
 import io.delta.kernel.test.MockSnapshotUtils.getMockSnapshot
 import io.delta.kernel.types.StringType
+import io.delta.kernel.types.StructType
 import io.delta.kernel.utils.FileStatus
 
 import org.scalatest.funsuite.AnyFunSuite
 
 class DeltaHistoryManagerSuite extends AnyFunSuite with MockFileSystemClientUtils {
 
+  // Helper function for non-catalog-managed tables (no staged commits)
```

Comment:
Since there are no default params in Java, but I didn't want to keep a second overloaded method def just for test-only usage, I had to add this test helper method. While I was at it, also refactored the boolean parameters to have defaults and only be defined when different. Sorry a lot of code change from this :(